src/apps/the-exchange-app/components/CardsSwap/CardsSwap.tsx

```diff
@@ -0,0 +1,113 @@
+import { useContext, useState } from 'react';
+import { CardPosition, SwapReceive } from '../../utils/types';
+import SwapReceiveCard from '../SwapReceiveCard/SwapReceiveCard';
+import SwapIcon from '../../images/arrow-swap-horizontal.png';
+import { SwapDataContext } from '../../context/SwapDataProvider';
+import DropdownTokenList from '../DropdownTokensList/DropdownTokenList';
+import usePillarSwapAssets from '../../hooks/usePillarSwapAssets';
+import { isEqual } from 'lodash';
+
+type CardPositionType = {
+    left: CardPosition;
+    right: CardPosition;
+}
+
+const CardsSwap = () => {
+    const {
+        setSwapTokenData,
+        isSwapOpen,
+        setIsSwapOpen,
+        setIsReceiveOpen,
+        isReceiveOpen,
+        swapChain,
+        receiveChain,
+        swapToken,
+        receiveToken,
+        setSwapChain,
+        setReceiveChain,
+        setSwapToken,
+        setReceiveToken,
+        amountSwap,
+        amountReceive,
+        setAmountReceive,
+        setAmountSwap,
+        setReceiveTokenData,
+    } = useContext(SwapDataContext);
+
+    const { getPillarSwapAssets } = usePillarSwapAssets();
+
+    const initialPosition: CardPositionType = {
+        left: CardPosition.LEFT,
+        right: CardPosition.RIGHT,
+    };
+
+    const [cardPosition, setCardPosition] = useState<CardPositionType>(initialPosition);
+
+    const swapCards = () => {
+        setCardPosition(prevPosition => ({
+            left: prevPosition.right,
+            right: prevPosition.left,
+        }));
+        setSwapChain(receiveChain);
+        setReceiveChain(swapChain);
+        setSwapToken(receiveToken);
+        setReceiveToken(swapToken);
+        setAmountSwap(amountReceive);
+        setAmountReceive(amountSwap);
+    };
+
+    const handleClick = async (position: CardPosition) => {
+        const type = position === CardPosition.LEFT ? SwapReceive.SWAP : SwapReceive.RECEIVE;
+        try {
+            const assets = await getPillarSwapAssets();
+            setSwapTokenData(assets);
+            setReceiveTokenData(assets);
+            if (type === SwapReceive.SWAP) {
+                setIsSwapOpen(true);
+            } else {
+                setIsReceiveOpen(true);
+            }
+        } catch (e) {
+            console.error(e);
+        }
+    };
+
+    const isInitialOrder = isEqual(cardPosition, initialPosition);
+
+    const renderCards = () => (
+        <div className='flex w-full gap-4 desktop:gap-8 justify-center'>
+            <SwapReceiveCard
+                onClick={() => handleClick(isInitialOrder ? cardPosition.left : cardPosition.right)}
```

Comment:
Handle click is a little bit generic, could we make it more specific please?

Reply:
Yes agreed, I changed it to: handleOpenTokenList